src/settings/factory_reset.c

```diff
@@ -0,0 +1,305 @@
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "card_operations.h"
+#include "coin_specific_data.h"
+#include "constant_texts.h"
+#include "flash_api.h"
+#include "flash_struct.h"
+#include "settings_api.h"
+#include "ui_core_confirm.h"
+#include "ui_screens.h"
+#include "ui_state_machine.h"
+#include "wallet_list.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+/**
+ * @brief Get the wallet list from two X1 cards
+ *
+ * @param wallet_list Pointer to buffer which will be filled with the wallet
+ * list of type wallet_list_t
+ * @return true If the card flow occurred successfully
+ * @return false If any terminal error occurred during the card flow
+ */
+static bool get_wallet_list_from_two_cards(wallet_list_t *wallet_list);
+
+/**
+ * @brief Get the common wallet list from the wallet list retrieved from two X1
+ * cards
+ *
+ * @param wallets_in_card1 Pointer to buffer filled with wallet list in card #1
+ * @param wallets_in_card2 Pointer to buffer filled with wallet list in card #2
+ * @param common_wallets Pointer to buffer which will be filled with the common
+ * wallets in card #1 and card #2
+ */
+static void get_common_wallet_list(wallet_list_t *wallets_in_card1,
+                                   wallet_list_t *wallets_in_card2,
+                                   wallet_list_t *common_wallets);
+
+/**
+ * @brief Get the uncommon wallet list between the X1 vault and two X1 cards
+ *
+ * @param wallets_in_vault Pointer to buffer filled with wallet list in X1 vault
+ * @param wallets_in_cards Pointer to buffer filled with common wallet list in
+ * two cards
+ * @param uncommon_wallets Pointer to buffer which will be filled with the
+ * wallets present in X1 vault but not in the two cards
+ */
+static void get_uncommon_wallet_list(wallet_list_t *wallets_in_vault,
```

Comment:
please see if we can reduce duplicity in logic

Reply:
I have tried by splitting the comparison logic and logic whether to add the wallet to the list or not as part of a callback.